### DISCUSSION TITLE
Pointing existing Logstash to Security Onion

### DISCUSSION THREAD

**Discussion by hacker0ni** [Security-Onion-Solutions/securityonion#5452] [Unsupported Versions] [answered]
I'm wondering if we could use an existing extrernal Logstash instance with parsing and point it to the ElasticSearch running on the manager node? Should I use Filebeat for this instead of Logstash? I got a distributed deployment with one of each node for now. My main goal is to gather logs on a machine and ship them to SO for centralization. These are mainly OSQuery & syslog logs. It looks like this is something that a lot of people are struggling with and I'd be happy to help if directed in the right way. Thanks in advance!

**Comment by dougburks** [comment]
Have you checked our documentation?
https://docs.securityonion.net/en/2.3/beats.html
https://docs.securityonion.net/en/2.3/osquery.html
https://docs.securityonion.net/en/2.3/syslog.html

**Comment by hacker0ni** [comment]
Thanks for the reply, I've been digging through the documentation for about a month now. The issue I'm having is there is no clear way on implementing custom Ingest Pipelines of ES. I can ship logs with Filebeat but at this current architecture of mine, it makes more sense to use Logstash to feed logs to Security Onion. Filebeat logs or Syslogs aren't being parsed/indexed by the ElasticSearch pipelines either, which is kind of weird. At the current state, there are too many directories on the appliance that I can't wrap my head around what is where and which files I'm supposed to edit to push a new ingest pipeline and make it be used if some conditions are met to properly parse & index data. Is there something I'm missing in the documentation?

**Reply by dougburks** [reply]
> The issue I'm having is there is no clear way on implementing custom Ingest Pipelines of ES.

Have you seen this part of the documentation?
https://docs.securityonion.net/en/2.3/elasticsearch.html#parsing

**Reply by n0th1nt0lOOz** [reply]
Hi @dougburks 
So if I use an external logstash and point output to the manager node (in distributed mode) without having the same ingest pipelines with the SO 's logstash pipelines, the ES on manager node still have ability to parse the log into right indexes that SO could use to analyze, right?

If my thinking is right, how can I find the way to authenticate between my external logstash and SO's elasticsearch?

I'm still very new to ELK and SO2. Hope you could give me instructions.
Thanks & Regards

**Comment by hacker0ni** [comment]
If I understood the documentation correctly, we can not use external Logstash instances to ship logs to SO am I correct, since there is no way to push logs directly to Redis on the SO? (I couldn't get the Lumberjack output plugin to work with SO, for this purpose)

If I used Filebeat to ship logs to SO, I would allow the endpoint using `so-allow -b`. I would need to put my custom ingest node pipeline into `/opt/so/saltstack/local/salt/elasticsearch/files/ingest/` and restart ElasticSearch. I would need to configure a new logstash pipeline and put it under `/opt/so/saltstack/local/salt/logstash/pipelines/config/custom/` and then add the following to the global.sls for manager processing: 

```
logstash:
  pipelines:
    manager:
      config:
        - so/0009_input_beats.conf
        - so/0010_input_hhbeats.conf
        - so/9999_output_redis.conf.jinja
        - custom/9999_output_custom.jinja
```
If I understand correctly, the 0009 config listens on port 5044 and adds the tag of "beat-ext" to incoming logs, then 9999 takes those and pushes them to the Redis queue, then 0010 takes them from the Redis and pushes it to ES. Then we have to use a custom output config (9999) to push them properly to ES.

This arises some questions:
1 - Should I edit 0009_input_beats.conf to add the tag I want, so that I can use that tag to direct the log to the correct pipeline on the custom Logstash output plugin (eg. 9999_output_custom.jinja)?
2 - In the custom ElasticSearch pipeline, do I need to use the pipeline "common" at the end to make my logs properly be indexed into Kibana/Hunt?
3 - How should we write the exemplary 9999_output_custom.jinja file to make sure these logs that are injected into SO can be properly parsed using the correct Ingest Pipelines and indexes?

**Reply by n0th1nt0lOOz** [reply]
Hope that ext logstash ship to SO could work. I have to share system logs with IT team but I don't want them to access the SO so I built a new ELK for them and make an ext logstash send output to 2 ES address - SO and IT log system

**Comment by hacker0ni** [comment, marked as answer]
See #5819
